### DISCUSSION TITLE
SF CLI Bulk Data Enhancements

### DISCUSSION THREAD

**Discussion by stuartgrieve** [forcedotcom/cli#2701] [Ideas]
Using the bulk data commands are far more favourable over using Data Loader, however, there's two features we're missing which means we have to split our approach between the two:

1. You can't specify a batch size
2. No option to run in Serial mode

It would be really cool to have parameters for these options

**Comment by mshanemc** [comment]
For 2, 

- the original command `force:data:bulk:upsert` has a `--serial` option.  That command uses bulk api v1.
- the new command `data bulk upsert` uses bulk api v2, which does not have a serial option, so the command doesn't either.  There's an idea for that here https://ideas.salesforce.com/s/idea/a0B8W00000GdYOBUA3/bulk-api-v2-support-for-serial-concurrencymode

**Comment by mshanemc** [comment]
For 1
the API doesn't let you change the batch size for a big uploaded CSV like we use.  

The way DataLoader works around this is to do the batching for you from the client side (splitting your csv and uploading it repeatedly to the `/batches` endpoint).  

the CLI currently just uploads the entire CSV to the API and lets it do its thing.

**Comment by stuartgrieve** [comment]
Thanks Shane!
